File: spring-data-requery/src/main/java/org/springframework/data/requery/repository/config/InspectionClassLoader.java
package org.springframework.data.requery.repository.config;

import org.springframework.instrument.classloading.ShadowingClassLoader;

/**
 * Disposable {@link ClassLoader} used to inspect user-code classes within an isolated class loader without preventing
 * class transformation at a later time.
 *
 * @author debop
 * @since 18. 6. 6
 */
public class InspectionClassLoader extends ShadowingClassLoader {

    public InspectionClassLoader(ClassLoader parent) {
        super(parent, true);
        excludePackage("org.springframework.");
    }
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/repository/config/InspectionClassLoader.kt
package org.springframework.data.requery.repository.config

import org.springframework.instrument.classloading.ShadowingClassLoader

/**
 * Disposable [ClassLoader] used to inspect user-code classes within an isolated class loader without preventing
 * class transformation at a later time.
 *
 * @author debop
 */
class InspectionClassLoader(parent: ClassLoader): ShadowingClassLoader(parent, true) {

    init {
        excludePackage("org.springframework.")
    }
}